# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadRequest.h
# Commit: a50f60a612d3
# Full Hash: a50f60a612d31de79369e59933242ee10a5d810f
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r 0ac6d9d26e2b -r a50f60a612d3 dom/script/ScriptLoadRequest.h
--- a/dom/script/ScriptLoadRequest.h	Thu Feb 17 15:57:48 2022 +0000
+++ b/dom/script/ScriptLoadRequest.h	Thu Feb 17 15:57:48 2022 +0000
@@ -59,7 +59,7 @@
   NS_DECL_CYCLE_COLLECTION_NATIVE_CLASS(ScriptFetchOptions)
 
   ScriptFetchOptions(mozilla::CORSMode aCORSMode,
-                     enum ReferrerPolicy aReferrerPolicy, Element* aElement,
+                     enum ReferrerPolicy aReferrerPolicy,
                      nsIPrincipal* aTriggeringPrincipal,
                      nsIGlobalObject* aWebExtGlobal);
 
@@ -341,7 +341,7 @@
   virtual ~DOMScriptLoadContext();
 
  public:
-  explicit DOMScriptLoadContext(ScriptLoadRequest* aRequest);
+  explicit DOMScriptLoadContext(Element* aElement, ScriptLoadRequest* aRequest);
 
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS(DOMScriptLoadContext)
@@ -456,6 +456,9 @@
   Atomic<Runnable*> mRunnable;  // Runnable created when dispatching off thread
                                 // compile. Tracked here so that it can be
                                 // properly released during cancellation.
+
+  nsCOMPtr<Element> mElement;
+
   RefPtr<ScriptLoadRequest> mRequest;
 
   // Non-null if there is a document that this request is blocking from loading.